### PR TITLE
remove initial token flag and use from the memory

### PR DESCRIPTION
**Pull request by vishalxebia**

### CODE REVIEW

**Review by TizianoPerrucci — CHANGES_REQUESTED** [review]
- we dont need to store a new flag, rather track if during renewal the method was called with the initial_token or with an action token (the one we make request with)
- rename the flag to `is_initial_token`
- update the http Authentication header only in `token_auth_session` and avoid to call it when creating a new token in `create_auth_token`

File: tests/core/test_splunk_helper.py [hunk 1]
```diff
@@ -290,20 +287,17 @@ def test_token_auth_session(self, mocked_decode_token):
         # load a token in memory for validation
         status = CheckData()
         status.data['http://testhost:8089token'] = 'memorytokenpresent'
-        # make initial_token_flag false to not create new token
-        initial_token_flag = False
         persistence_check_name = "splunk_metric"
+        status.persist()
```
Comment (TizianoPerrucci) [inline tests/core/test_splunk_helper.py]: do we still need the persist here ?

File: tests/core/test_splunk_helper.py [hunk 2]
```diff
@@ -320,22 +314,19 @@ def test_token_auth_session_need_renewal_initial_token(self, mocked_decode_token
...
-        token_flag = helper.token_auth_session(auth, config.base_url, status, initial_token_flag,
-                                               persistence_check_name)
-        self.assertFalse(token_flag)
+        helper.token_auth_session(auth, config.base_url, status, persistence_check_name)
+
         # Header should be updated with the new token
         expected_header = helper.requests_session.headers.get("Authorization")
         self.assertEqual(expected_header, "Bearer {}".format(new_token))
-        # persistence data will have new token as well
+        # persistence data will have new token updated and token flag as False
```
Comment (TizianoPerrucci) [inline tests/core/test_splunk_helper.py]: update comment

**Review by TizianoPerrucci — CHANGES_REQUESTED** [review]
I like the changes. Just 2 very small things

File: tests/core/test_splunk_helper.py (hunk 1 above)
Comment (vishalxebia) [reply tests/core/test_splunk_helper.py]: removed.

File: tests/core/test_splunk_helper.py (hunk 2 above)
Comment (vishalxebia) [reply tests/core/test_splunk_helper.py]: updated the comment.

**Review by TizianoPerrucci — APPROVED** [review]

**Pull request merged**